core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -231,6 +234,41 @@ public ClientFactoryBuilder http2MaxFrameSize(int http2MaxFrameSize) {
     }
 
     /**
+     * Sets the maximum length of an HTTP/1 response initial line.
+     */
+    public ClientFactoryBuilder maxHttp1InitialLineLength(int maxHttp1InitialLineLength) {
+        checkArgument(maxHttp1InitialLineLength < 0,
+                "maxHttp1InitialLineLength: %s (expected: >= 0)",
+                maxHttp1InitialLineLength);
+        this.maxHttp1InitialLineLength = maxHttp1InitialLineLength;
+        return this;
+    }
+
+    /**
+     * Sets the maximum length of all headers in an HTTP/1 response.
+     */
+    public ClientFactoryBuilder maxHttp1HeaderSize(int maxHttp1HeaderSize) {
+        checkArgument(maxHttp1HeaderSize < 0,
+                "maxHttp1HeaderSize: %s (expected: >= 0)",
+                maxHttp1HeaderSize);
+        this.maxHttp1HeaderSize = maxHttp1HeaderSize;
+        return this;
+    }
+
+    /**
+     * Sets the maximum length of each chunk in an HTTP/1 response content.
+     * The content or a chunk longer than this value will be split into smaller chunks
+     * so that their lengths never exceeds it.
+     */
+    public ClientFactoryBuilder maxHttp1ChunkSize(int maxHttp1ChunkSize) {
+        checkArgument(maxHttp1ChunkSize < 0,
```

Comment:
`<` -> `=>`?

Reply:
How about extract this to validateNonNegative as well?

Reply:
How about adding some test that uses `ServerRule` and custom `ClientFactory` to make sure this code path is tested?

Reply:
.. such as sending a request with very long path and headers and receiving a response with very long path and headers.

Reply:
@minwoox 
Yes. I think you are right.
Thank you for finding what I missed.